src/api/auth/src/token.js

```diff
@@ -2,19 +2,32 @@ const jwt = require('jsonwebtoken');
 
 const { JWT_ISSUER, JWT_AUDIENCE, SECRET, JWT_EXPIRES_IN } = process.env;
 
-function createToken(subject) {
-  // TODO - figure out all the various claims we need to use
+/**
+ * Create a JWT token for the user, and add the 'admin' role if requested.
+ * @param {string} email the email of this user. This will be the sub claim
+ * @param {string} name a name suitable for display for this user.
+ * @param {Array<string>} roles an array of roles for this user
+ * @returns {string} the JWT for this user
+ */
+function createToken(email, name, roles) {
   const payload = {
     // The token is issued by us (e.g., this server)
     iss: JWT_ISSUER,
     // It is intended for the services running at this api origin
     aud: JWT_AUDIENCE,
-    // The subject of this token, the user
-    sub: subject,
-    // TODO: role info (e.g., admin)
+    // The subject of this token, the user's nameID (i.e., their Seneca email address)
+    sub: email,
+    // The user's display name
+    name,
+    // User roles.  We have 4 currently:
+    //   1. seneca (authenticated with Seneca's SSO)
+    //   1. telescope (authenticated Telescope user)
+    //   2. admin (authenticated Telescope user with isAdmin=true in Firebase)
+    //   3. service (a Telescope microservice, see createServiceToken() in Satellite)
+    roles,
   };
 
-  const options = { expiresIn: JWT_EXPIRES_IN || '1h' };
+  const options = { expiresIn: JWT_EXPIRES_IN || '3h' };
```

Comment:
Is there a way for us to extend this for PWA? I think most mobile apps let you stay logged in for a longer period of time than websites

Reply:
There's some risk in having tokens be long-lived, since they are effectively passwords that anyone can re-use without having to prove themselves.  We can extend this, for sure, but for how long?  With systems that do cookie based sessions on the server, you can have a login live forever, and not risk much.  With tokens that a browser holds, it's harder.  We could look into doing some kind of "refresh" token later, to let you renew your login.

Question: what would you do while logged in for longer?  The only time you need to log in is when you create or modify your account.

Reply:
Understandable. I was just thinking if there is a safe way to let users stay logged in while using PWA. To answer your question, that's true, but only for now ;)